src/core/obj.js

```diff
@@ -2172,10 +2173,25 @@ var XRef = (function XRefClosure() {
           );
         }
         nums[i] = num;
+        offsets[i] = offset;
       }
+
+      const start = (stream.start || 0) + first;
       const entries = new Array(n);
       // read stream objects for cache
       for (let i = 0; i < n; ++i) {
+        const length = i < n - 1 ? offsets[i + 1] - offsets[i] : undefined;
+        if (length < 0) {
+          throw new FormatError("invalid offset in the ObjStm stream");
+        }
+        parser = new Parser({
+          lexer: new Lexer(
+            stream.makeSubStream(start + offsets[i], length, stream.dict)
+          ),
+          xref: this,
+          allowStreams: true,
+        });
+
         const obj = parser.getObj();
         entries[i] = obj;
         // The ObjStm should not contain 'endobj'. If it's present, skip over it
```

Comment:
This will obviously require testing, but given that you're now *replacing* the parser for each loop iteration my question is if the `endobj` special-case here still necessary with this patch?

Reply:
I looked into `bug1037816.pdf` and yes we have these `endobj` as  garbage at the end of the chunk.
So having a new Parser with the correct starting offset helps to skip those.
So it shouldn't be a regression to remove that stuff.

Reply:
I've just tested all of the files listed in https://github.com/mozilla/pdf.js/blob/ba76dd4000c0efa570558dd56d551b773bb4e856/src/core/obj.js#L2182 and all of them still work fine with this code removed :-)

---

Can you please also change https://github.com/mozilla/pdf.js/blob/ba76dd4000c0efa570558dd56d551b773bb4e856/test/test_manifest.json#L2040 to an `eq` test instead, as part of this patch, since it's not generally *guaranteed* that just a `load` test catches all kind of /XRef-parsing errors?